fedimint-core/src/db/mod.rs

```diff
@@ -214,6 +225,74 @@ impl Database {
                 .expect("db autocommit retry counter overflowed");
         } // end of loop
     }
+
+    /// Waits for key to be notified.
+    ///
+    /// Calls the `f` on every notifications. Return the value when `f` returns
+    /// a `Some(T)`
+    pub async fn wait_key_check<'a, K, T>(
```

Comment:
Needs to handle shutdowns. How to do this? should I just add argument `tg_handle: ..`

Reply:
Why would it need special handling here? The caller can just put a `select!` around the main future/loop and the stop signal at a higher level assuming this function is cancellation safe. I think we should handle shutdowns at the highest possible level.